lib/vsc/install/shared_setup.py

```diff
@@ -174,6 +187,123 @@ def run(self):
         orig_bdist_rpm.run(self)
 
 
+# private class variables to communicate
+# between VscScanningLoader and VscTestCommand
+# stored in __builtin__ because the .run_tests
+# reload and cleanup in the modules
+import __builtin__
+if not hasattr(__builtin__,'__test_filter'):
+    setattr(__builtin__, '__test_filter',  {
+        'module': None,
+        'function': None,
+        'allowmods': [],
+    })
+
+
+def filter_testsuites(testsuites):
+    """(Recursive) filtering of (suites of) tests"""
+    test_filter = getattr(__builtin__,'__test_filter')['function']
+
+    res = type(testsuites)()
+
+    for t in testsuites:
+        # t is either a test or testsuite of more tests
+        if isinstance(t, TestSuite):
+            res.addTest(filter_testsuites(t))
+        else:
+            if re.search(test_filter, t._testMethodName):
+                res.addTest(t)
+    return res
+
+
+class VscScanningLoader(ScanningLoader):
+    def loadTestsFromModule(self, module):
+        """Allow for filter"""
```

Comment:
needs a better docstring, should provide a better description of what `loadTestsFromModule` does/returns?